# Lab | Hypothesis Testing

**Objective**

Welcome to the Hypothesis Testing Lab, where we embark on an enlightening journey through the realm of statistical decision-making! In this laboratory, we delve into various scenarios, applying the powerful tools of hypothesis testing to scrutinize and interpret data.

From testing the mean of a single sample (One Sample T-Test), to investigating differences between independent groups (Two Sample T-Test), and exploring relationships within dependent samples (Paired Sample T-Test), our exploration knows no bounds. Furthermore, we'll venture into the realm of Analysis of Variance (ANOVA), unraveling the complexities of comparing means across multiple groups.

So, grab your statistical tools, prepare your hypotheses, and let's embark on this fascinating journey of exploration and discovery in the world of hypothesis testing!

**Challenge 1**

In this challenge, we will be working with pokemon data. The data can be found here:

- https://raw.githubusercontent.com/data-bootcamp-v4/data/main/pokemon.csv

In [7]:
#libraries
import pandas as pd
import scipy.stats as st
import numpy as np



In [8]:
df = pd.read_csv("https://raw.githubusercontent.com/data-bootcamp-v4/data/main/pokemon.csv")
print(df.columns.tolist())

['Name', 'Type 1', 'Type 2', 'HP', 'Attack', 'Defense', 'Sp. Atk', 'Sp. Def', 'Speed', 'Generation', 'Legendary']


- We posit that Pokemons of type Dragon have, on average, more HP stats than Grass. Choose the propper test and, with 5% significance, comment your findings.

In [9]:
#code here
dragon_hp = df[df['Type 1'] == 'Dragon']['HP']
grass_hp = df[df['Type 1'] == 'Grass']['HP']

# Welch's t-test (recommended)
t_stat, p_two_sided = st.ttest_ind(
    dragon_hp,
    grass_hp,
    equal_var=False
)

# Convert to one-sided p-value
if t_stat > 0:
    p_one_sided = p_two_sided / 2
else:
    p_one_sided = 1 - p_two_sided / 2

print("t-statistic:", t_stat)
print("p-value:", p_one_sided)

t-statistic: 3.3349632905124063
p-value: 0.0007993609745420597


Since the p-value (0.000799) is less than the significance level of 0.05, we reject the null hypothesis. There is sufficient statistical evidence to conclude that Dragon-type Pokémon have a higher average HP stat than Grass-type Pokémon.

- We posit that Legendary Pokemons have different stats (HP, Attack, Defense, Sp.Atk, Sp.Def, Speed) when comparing with Non-Legendary. Choose the propper test and, with 5% significance, comment your findings.


In [10]:
#code here
from scipy.stats import ttest_ind

stats_cols = ['HP', 'Attack', 'Defense', 'Sp. Atk', 'Sp. Def', 'Speed']

legendary = df[df['Legendary'] == True]
non_legendary = df[df['Legendary'] == False]

for stat in stats_cols:
    t_stat, p_value = ttest_ind(
        legendary[stat],
        non_legendary[stat],
        equal_var=False  # Welch's t-test
    )

    print(f'{stat}:')
    print(f'  t-statistic = {t_stat:.4f}')
    print(f'  p-value = {p_value:.6f}')

    if p_value < 0.05:
        print('  Significant difference\n')
    else:
        print('  No significant difference\n')

HP:
  t-statistic = 8.9814
  p-value = 0.000000
  Significant difference

Attack:
  t-statistic = 10.4381
  p-value = 0.000000
  Significant difference

Defense:
  t-statistic = 7.6371
  p-value = 0.000000
  Significant difference

Sp. Atk:
  t-statistic = 13.4174
  p-value = 0.000000
  Significant difference

Sp. Def:
  t-statistic = 10.0157
  p-value = 0.000000
  Significant difference

Speed:
  t-statistic = 11.4750
  p-value = 0.000000
  Significant difference



Using independent two-sample t-tests at the 5% significance level, all six stats (HP, Attack, Defense, Sp. Atk, Sp. Def, and Speed) showed statistically significant differences between Legendary and Non-Legendary Pokémon (all p-values < 0.001). Therefore, we conclude that Legendary Pokémon have significantly different overall stats compared with Non-Legendary Pokémon.

**Challenge 2**

In this challenge, we will be working with california-housing data. The data can be found here:
- https://raw.githubusercontent.com/data-bootcamp-v4/data/main/california_housing.csv

In [14]:
df = pd.read_csv("https://raw.githubusercontent.com/data-bootcamp-v4/data/main/california_housing.csv")
df.head()

,longitude,latitude,housing_median_age,total_rooms,total_bedrooms,population,households,median_income,median_house_value
0,-114.31,34.19,15.0,5612.0,1283.0,1015.0,472.0,1.4936,66900.0
1,-114.47,34.40,19.0,7650.0,1901.0,1129.0,463.0,1.8200,80100.0
2,-114.56,33.69,17.0,720.0,174.0,333.0,117.0,1.6509,85700.0
3,-114.57,33.64,14.0,1501.0,337.0,515.0,226.0,3.1917,73400.0
4,-114.57,33.57,20.0,1454.0,326.0,624.0,262.0,1.9250,65500.0


**We posit that houses close to either a school or a hospital are more expensive.**

- School coordinates (-118, 34)
- Hospital coordinates (-122, 37)

We consider a house (neighborhood) to be close to a school or hospital if the distance is lower than 0.50.

Hint:
- Write a function to calculate euclidean distance from each house (neighborhood) to the school and to the hospital.
- Divide your dataset into houses close and far from either a hospital or school.
- Choose the propper test and, with 5% significance, comment your findings.
 

In [12]:
def euclidean_distance(x1, y1, x2, y2):
    return np.sqrt((x1 - x2)**2 + (y1 - y2)**2)

In [15]:
df['dist_school'] = euclidean_distance(
    df['longitude'],
    df['latitude'],
    -118,
    34
)

df['dist_hospital'] = euclidean_distance(
    df['longitude'],
    df['latitude'],
    -122,
    37
)

df['Close'] = (
    (df['dist_school'] < 0.5) |
    (df['dist_hospital'] < 0.5)
)

close_prices = df[df['Close']]['median_house_value']
far_prices = df[~df['Close']]['median_house_value']

t_stat, p_value = ttest_ind(
    close_prices,
    far_prices,
    equal_var=False
)

print("t-statistic:", t_stat)
print("p-value:", p_value)

t-statistic: 37.992330214201516
p-value: 3.0064957768592614e-301


Appropriate Test

An independent two-sample t-test (Welch's t-test) was used to compare house prices between neighborhoods that are close to either a school or a hospital and those that are not.

Hypotheses

H₀: The average house price is the same for houses close to a school or hospital and houses farther away.
H₁: Houses close to a school or hospital have higher average prices.

Results

t-statistic = 37.992
p-value = 3.01 × 10⁻³⁰¹

Conclusion

Since the p-value is far below the 5% significance level (0.05), we reject the null hypothesis. There is extremely strong statistical evidence that houses located close to either a school or a hospital have higher average prices than houses located farther away. Therefore, the data supports the claim that proximity to a school or hospital is associated with more expensive houses.